server/dirstore_test.go

```diff
@@ -436,28 +437,27 @@ func TestExpiration(t *testing.T) {
 		createTestAccount(t, dirStore, expSec, accountKey)
 	}
 
-	h := dirStore.Hash()
+	hBegin := dirStore.Hash()
+	account(100)
+	hNoExp := dirStore.Hash()
+	require_NotEqual(t, hBegin, hNoExp)
+	account(1)
+	nh2 := dirStore.Hash()
+	require_NotEqual(t, hNoExp, nh2)
+	assertStoreSize(t, dirStore, 2)
 
-	for i := 1; i <= 3; i++ {
-		account(i)
-		nh := dirStore.Hash()
-		require_NotEqual(t, h, nh)
-		h = nh
-	}
-	time.Sleep(500 * time.Millisecond)
-	for i := 3; i > 0; i-- {
+	failAt := time.Now().Add(4 * time.Second)
+	for time.Now().Before(failAt) {
```

Comment:
So this makes this test run for at least 4 seconds?

Reply:
No, as you can see there is a return in the loop once the assertion is done. The 4sec is the max before reporting error

Reply:
So should only run 100ms from first sleep etc in optimum case yes?

Reply:
about a second.
In part expiration in the jwt has to be bigger than issue at. and so we have to do rounding when issuing the jwt.